fastavro/util.py

```diff
@@ -0,0 +1,72 @@
+# cython: auto_cpdef=True
+'''Utility functions'''
+
+import sys
+from functools import wraps
+
+if sys.version_info < (3,):
+    INTEGER_TYPES = (int, long,)  # flake8: noqa
+else:
+    INTEGER_TYPES = (int,)
+
+
+def push_path(path, name):
+    if path is None:
+        return [name]
+    else:
+        path.append(name)
+        return path
+
+
+def pop_path(path):
+    if path is None:
+        return None
+    else:
+        path.pop()
+        return path
+
+
+def path_string(path):
+    def format_item(item):
```

Comment:
Regarding the need for push_path and family: setting default parameters to mutable objects like list   leads to all kinds of unexpected behavior. Making the path a required parameter would fix this, though, but it changes the interface to all the read_* and write_* functions, which I didn't want to do.